spec/System/HapistranoConfigSpec.hs

```diff
@@ -36,6 +39,27 @@ spec =
         Yaml.loadYamlSettings ["fixtures/git_repository_config.yaml"] [] Yaml.useEnv
           >>= (`shouldBe` defaultConfiguration)
 
+    describe "Config From JSON instances" $ do
+      context "when contains unique targets" $
+        it "parses all unique targets" $
+          let eiHosts = fmap targetHost <$> configHosts <$> A.eitherDecode content
+              content =
+                "{\"targets\":[{\"host\":\"user@app1.com\"},{\"host\":\"user@app2.com\"}],\"deploy_path\":\"/tmp\",\"local_directory\":\"/\"}"
+          in case eiHosts of
+              Left msg -> fail msg
+              Right hosts ->
+                hosts `shouldBe` ["user@app1.com", "user@app2.com"]
```

Comment:
@CristhianMotoche I think this pattern match is not necessary, if you use `shouldBe` it should take care of printing the expected and got value, in the case of a `Left` it should include the error message

Reply:
Good catch @sestrella !!!